src/Xamarin.Android.Build.Tasks/Tasks/ResolveLibraryProjectImports.cs

```diff
@@ -431,6 +442,11 @@ public override bool RunTask ()
 					resolvedAssetDirectories.Add (new TaskItem (Path.GetFullPath (assetsDir), new Dictionary<string, string> {
 						{ OriginalFile, aarFullPath },
 					}));
+				if (File.Exists (proguardFile) && IsApplication) {
```

Comment:
The main reason I did this was so that when we build a library project we don't even call proguard/r8 so its not really needed. But we do use it for Apps, so restricting it to apps only seemed logical.